File: python/bc2626.py
while True:
    try:
        msg = {
            0: "Putz vei, o Leo ta demorando muito pra jogar...",
            1: "Os atributos dos monstros vao ser inteligencia, sabedoria...",
            2: "Iron Maiden's gonna get you, no matter how far!",
            3: "Urano perdeu algo muito precioso..."
        }
        res = -1
        line = input().split(" ")

        dodo = line[0]
        leo = line[1]
        pepper = line[2]

        if dodo == leo and leo == pepper: res = 0
        else:
            if (dodo == "papel" and leo == "pedra" and pepper == "pedra") or \
               (dodo == "tesoura" and leo == "papel" and pepper == "papel") or \
               (dodo == "pedra" and leo == "tesoura" and pepper == "tesoura"):  res = 1
            elif (leo == "papel" and dodo == "pedra" and pepper == "pedra") or \
                 (leo == "tesoura" and dodo == "papel" and pepper == "papel") or \
                 (leo == "pedra" and dodo == "tesoura" and pepper == "tesoura"): res = 2
            elif (pepper == "papel" and leo == "pedra" and dodo == "pedra") or \
                 (pepper == "tesoura" and leo == "papel" and dodo == "papel") or \
                 (pepper == "pedra" and leo == "tesoura" and dodo == "tesoura"): res = 3

        print(f"{msg[res]}")
    except EOFError:
        break

File: cpp/bc2626.cpp
#include <iostream>
#include <string>
#include <sstream>

using namespace std;

int main()
{
	string line, dodo, leo, pepper, msg, token;
	int res;
	
	while(getline(cin, line))
	{	
		istringstream iss(line);
		
		iss >> token;
		dodo = token;
		iss >> token;
		leo = token;
		iss >> token;
		pepper = token;
		
		res = -1;
		
		if(dodo == leo && leo == pepper) res = 0;
		else
		{
			if((dodo == "tesoura" && leo == "papel" && pepper == "papel") ||
			(dodo == "pedra" && leo == "tesoura" && pepper == "tesoura") ||
			(dodo == "papel" && leo == "pedra" && pepper == "pedra")) res = 1;
			else if((leo == "tesoura" && dodo == "papel" && pepper == "papel") ||
			(leo == "pedra" && dodo == "tesoura" && pepper == "tesoura") ||
			(leo == "papel" && dodo == "pedra" && pepper == "pedra")) res = 2;
			else if((pepper == "tesoura" && leo == "papel" && dodo == "papel") ||
			(pepper == "pedra" && leo == "tesoura" && dodo == "tesoura") ||
			(pepper == "papel" && leo == "pedra" && dodo == "pedra")) res = 3;
			else res = 0;
		}
		
		switch(res)
		{
			case 0: msg = "Putz vei, o Leo ta demorando muito pra jogar..."; break;
			case 1: msg = "Os atributos dos monstros vao ser inteligencia, sabedoria..."; break;
			case 2: msg = "Iron Maiden's gonna get you, no matter how far!"; break;
			case 3: msg = "Urano perdeu algo muito precioso..."; break;
		}
		
		cout << msg << endl;
		
		cin.clear();
	}
}
